blog/_src/posts/2017-02-26-type-tailoring.scrbl

```diff
@@ -0,0 +1,461 @@
+#lang scribble/manual
+@require[
+  pict
+  scribble/example
+  (rename-in scribble/manual
+    [racket scrbl:racket])
+  (for-syntax
+    racket/base)
+  (for-label
+    racket/base
+    racket/list
+    syntax/parse
+    (only-in racket/port
+      with-input-from-string)
+    (only-in typed/racket/base
+      :))
+]
+
+@(define tsam-url
+   "http://www.ccs.neu.edu/home/stchang/popl2017/")
+@(define tt-preprint
+   "http://www.ccs.neu.edu/home/types/resources/type-tailoring.pdf")
+@(define trivial-docs
+   "http://docs.racket-lang.org/trivial/index.html")
+@(define mh-popl-1988
+   "http://www.lfcs.inf.ed.ac.uk/reports/87/ECS-LFCS-87-42/ECS-LFCS-87-42.pdf")
+@(define growing-a-language
+   "https://www.youtube.com/watch?v=_ahvzDzKdB0")
+@(define tr
+   "http://docs.racket-lang.org/ts-reference/index.html")
+@(define w-popl-2017
+   "http://www.cis.upenn.edu/~sweirich/talks/popl17.pdf")
+@(define dth-popl17
+   "https://github.com/sweirich/dth/tree/master/popl17")
+@(define trivial-pkg-url
+   "https://pkgd.racket-lang.org/pkgn/package/trivial")
+@(define trivial-pkg
+   @hyperlink[trivial-pkg-url]{@tt{trivial}})
+
+@(define (section* . args)
+   (apply section args #:style 'unnumbered))
+@(define (subsection* . args)
+   (apply subsection args #:style 'unnumbered))
+@(define (subsubsection* . args)
+   (apply subsubsection args #:style 'unnumbered))
+@(define-syntax (racket stx)
+   (syntax-case stx ()
+     [(_ x) #'(elem #:style "RktWrap" (scrbl:racket x))]))
+
+@; =============================================================================
+
+Title: Type Tailoring
+Date: 2017-02-26T16:35:47
+Tags:
+
+@emph{posted by Ben Greenman}
+
+Type tailoring is a technique that library authors can use to build
+ @emph{domain-specific} type checkers to extend the type checking
+ provided by the host language.
+The idea is simple: give library authors control over how uses of their API
+ expand from source code to a typed intermediate representation.
+Using the @hyperlink[tsam-url]{@emph{Type Systems as Macros}} approach to building
+ typed language, the implementation is also straightforward, and the new typing
+ rules can be proven sound relative to the host language.
+
+<!-- more -->
+
+@section*{Background: Programming the Type Elaborator}
+
+Most typed languages include both a type checker and a @emph{type elaborator}.
```

Comment:
I don't think this is true about "most", regardless of how you draw the boundary.